doc/api/process.markdown

```diff
@@ -449,6 +554,97 @@ console.log(process.env.TEST);
 // => undefined
 ```
 
+## process.emitWarning(warning)
+
+The `process.emitWarning(warning)` method can be used to emit custom 
+or application specific process warnings. These can be listened for by adding
+a handler to the [`process.emit('warning')`][process_warning] event.
+
+* `warning` {String|Object|Error} The warning to issue
+
+The `warning` argument can be either a string, an Error object, or a simple 
+JavaScript object with a `name` and `message` property.
+
+If `warning` is passed as a string, the warning name is set to 'Warning' and
+the provided string is used as the message.
+
+```js
+// Emit a warning using a string...
+process.emitWarning('Warning! Something happened!');
+  // Emits: Warning: (node:56338) Warning! Something happened!
+```
+If `warning` is passed as an object, `process.emitWarning()` will look for a
+`name` and `message` property. If `name` is not specified, 'Warning' is used
+as the default. If a `message` property is not specified, a `TypeError` will
+be thrown.
+
+```
+// Emit a warning using an object...
+process.emitWarning({
+  name: 'CustomWarning', 
+  message: 'Warning! Something Happened!'
+})
+  // Emits: CustomWarning: (node:56338) Warning! Something Happened!
+```
+
+In each of the previous examples, an `Error` object is generated and passed 
+through to the [`process.emit('warning')`][process_warning] event.
+
+```
+process.on('warning', (warning) => {
+  console.log(warning.name);
+  console.log(warning.message);
+  console.log(warning.stack);
+});
+```
+
+If `warning` is passed as an `Error` object, it will be passed through to the
+`process.emit('warning')` event handler unmodified:
+
+```
+// Emit a warning using an Error object...
+const util = require('util');
+
+function CustomWarning(message) {
```

Comment:
If we do `Error.call(this)` probably means we don't need to do `Error.captureStackTrace(this)` - right?